pkg/backup/backup_compaction.go

```diff
@@ -932,6 +1021,58 @@ func compactionJobDescription(details jobspb.BackupDetails) (string, error) {
 	return fmtCtx.CloseAndGetString(), nil
 }
 
+// compactionHeuristic is a function that given a backup chain, returns the start
+// and end timestamps of compaction based on some heuristic.
+type compactionHeuristic func(
```

Comment:
nit: perhaps we move the compaction picking code to it's own file `compaction_policy.go`?